### ISSUE TITLE
 Error of riru on Android 12

### ISSUE THREAD

**Issue by lyy0709** [RikkaApps/Riru#278]
Please report bugs of Riru itself. Always use **the latest debug version** of Riru, and **make sure any other modules are disabled**.

**Reports not meet the requirements will be immediately closed**.

**Information:**

- Except Riru itself, all other Magisk modules are disabled: [No]
- Riru version: v26.1.2.r505.c20529bced
- Magisk version: f822ca5b(23001)
- Android version:android 12
- Device: oneplus 9pro

**Logs:**

Related **full logcat**. For example, if you have a "Riru module not loaded" problem, a full boot log will be helpful.
Here is my long about riru:
 05-16 05:07:07.875   828   830 I : * Running module post-fs-data scripts
 05-16 05:07:07.876   836   836 I : riru-core: exec [post-fs-data.sh]
 05-16 05:07:07.891   836   836 I : riru_lsposed: exec [post-fs-data.sh]
 05-16 05:07:07.894   828   835 D : proc_monitor: nothing to monitor, wait for signal
 05-16 05:07:07.897   836   836 I : riru_thanox: exec [post-fs-data.sh]
 05-16 05:07:07.898   828   830 I : * Loading modules
 05-16 05:07:07.898   828   830 I : riru-core: loading [system.prop]
 05-16 05:07:07.898   828   830 D : resetprop: Parse prop file [/dev/zxh7u/.magisk/modules/riru-core/system.prop]
 05-16 05:07:07.898   828   830 D : resetprop: delete prop [ro.dalvik.vm.native.bridge]
 05-16 05:07:07.898   828   830 D : resetprop: create prop [ro.dalvik.vm.native.bridge]: [libriruloader.so] by modifying prop data structure
 05-16 05:07:07.898   828   830 I : riru-core: loading mount files
 05-16 05:07:07.898   828   830 I : riru_lsposed: loading [system.prop]

**Describe the bug:**

A clear and concise description of what the bug is Riru failed to load,including the magisk is broken on the device,post-fs-data.sh fails to [run]([url](url)) .

**Additional context:**

**Comment by RikkaW** [comment]
Open a new issue if you can provide full boot log rather than Magisk log.

**Comment by EugenNY** [comment]
@RikkaW I am having the same problem on 2 different Moto phones (G5 and G6)
```
 post-fs-data.sh fails to run. Magisk is broken on this device.
```

If i run `post-fs-data.sh` manually it kind of works after a few attempts.
But I can't get LSPosed to work.

- latest Magisk
- Android 8.1 and Android 9

how to create a "full boot log"? logcat will give me only last few seconds.

**Comment by RikkaW** [comment]
@EugenNY 

> post-fs-data.sh fails to run. Magisk is broken on this device.

The post-fs-data script of the Riru module does not run. This means the module function of Magisk is broken on your device. It could be a Magisk problem or you have installed some low-quality modules which stuck the post-fs-data process.

> how to create a "full boot log"

Set log buffer size to the max in developer settings and start logcat ASAP.

Or, use this module https://t.me/hzrtools/10 (but considering the post-fs-data script does not run on your device, it may not work).

**Comment by EugenNY** [comment]
thanks for the quick response @RikkaW. I will try to use logcat as you suggested.
as I have pointed out, both phones have latest (and official) Magisk v23.0.
there are no other modules installed. 
I tried multiple times to install after flashing a fresh stock ROM. 
I think I got it to run with Android 8.0 on one device, but LSPosed only supports 8.1+
with the Android 9 and 8.1 stock ROMs I keep getting the same error.

on both phones, the following 2 directories are empty
`/data/adb/post-fs-data.d`
`/data/adb/service.d/'

I also noticed that `/data/adb/modules/riru-core/post-fs-data.sh` and `service.sh` didn't have a `+x` so I did a `chmod +x` for both and rebooted, but that didn't resolve anything. 

Something is wrong in how the module is installed or initialized.

```
 * Initializing Magisk environment      
 * Running post-fs-data.d scripts       
 * Running module post-fs-data scripts  
 riru-core: exec [post-fs-data.sh]      
 * Loading modules                      
 riru-core: loading [system.prop]       
 riru-core: loading mount files         
 ** late_start service mode running     
 * Running service.d scripts            
 * Running module service scripts       
 riru-core: exec [service.sh]           
 ** boot_complete triggered             
```
[magisk_log_2022-01-07.log](https://github.com/RikkaApps/Riru/files/7827982/magisk_log_2022-01-07.log)
